### ISSUE TITLE
InvalidArgumentError (see above for traceback): No OpKernel was registered to support Op 'MklConv2DWithBias' with these attrs.

### ISSUE THREAD

**Issue by tanhe123** [tensorflow/tensorflow#21145]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: mac, debian 8.7
- **Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device**:
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: 1.1.0
- **Python version**: Python 2.7.13
- **Bazel version (if compiling from source)**: 0.4.2
- **GCC/Compiler version (if compiling from source)**: gcc version 4.9.2 (Debian 4.9.2-10)
- **CUDA/cuDNN version**:
- **GPU model and memory**:
- **Exact command to reproduce**:

tf_env_collect.sh:

```
== cat /etc/issue ===============================================
Linux 4be8a8788f34 4.9.60-linuxkit-aufs #1 SMP Mon Nov 6 16:00:12 UTC 2017 x86_64 GNU/Linux
VERSION_ID="8"
VERSION="8 (jessie)"

== are we in docker =============================================
Yes

== compiler =====================================================
c++ (Debian 4.9.2-10) 4.9.2
Copyright (C) 2014 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.


== uname -a =====================================================
Linux 4be8a8788f34 4.9.60-linuxkit-aufs #1 SMP Mon Nov 6 16:00:12 UTC 2017 x86_64 GNU/Linux

== check pips ===================================================
intel-numpy            1.13.3.10
numpy                  1.14.5
protobuf               3.6.0
tensorflow             1.1.0

== check for virtualenv =========================================
False

== tensorflow import ============================================
tf.VERSION = 1.1.0
tf.GIT_VERSION = v1.1.0-0-g1ec6ed5
tf.COMPILER_VERSION = v1.1.0-0-g1ec6ed5
Sanity check: array([1], dtype=int32)

== env ==========================================================
LD_LIBRARY_PATH /code/:/code//lib:/usr/local/lib
DYLD_LIBRARY_PATH is unset

== nvidia-smi ===================================================
/tensor/tensorflow/tools/tf_env_collect.sh: line 105: nvidia-smi: command not found

== cuda libs  ===================================================
```

### Describe the problem
I posted a stackoverflow issue: https://stackoverflow.com/questions/51530592/invalidargumenterror-see-above-for-traceback-no-opkernel-was-registered-to-su

### How to Generate the problem

I found a way to reproduce this problem.

1. download tensorflow zip and unzip. 
  https://drive.google.com/file/d/1k616LEvgTUXpHuX6Fz7EDXukaG5tGZwQ/view

2. mount tensorflow folder to container:
  docker run --rm -it -v $(pwd):/code econtal/numpy-mkl  bash

3. create the test script:

  ```
  $ cd /code
  $ vi test_conv2d.py

  # demo from https://www.jianshu.com/p/a70c1d931395
  import tensorflow as tf 
  import tensorflow.contrib.slim as slim
  import numpy as np
 
  x1 = tf.ones(shape=[1, 64, 64, 3]) 
  w = tf.fill([5, 5, 3, 64], 1)
  # print("rank is", tf.rank(x1))

  # x1 = tf.cast(x1, tf.float32)  
  w = tf.cast(w, tf.float32)

  print('-----debugging-----')
  print(type(x1))
  print(x1.dtype.base_dtype)

  print(type(w))
  print(w.dtype.base_dtype)
  print('-------------------')

  # x1 = tf.cast(x1, tf.float16)

  y1 = tf.nn.conv2d(x1, w, strides=[1, 1, 1, 1], padding='SAME')
  y2 = slim.conv2d(x1, np.float32(64.0), np.array([5.0, 5.0], dtype=np.float32), 
  weights_initializer=tf.ones_initializer, padding='SAME')

  with tf.Session() as sess: 
      sess.run(tf.global_variables_initializer()) 
      y1_value,y2_value,x1_value=sess.run([y1,y2,x1])
      print("shapes are", y1_value.shape, y2_value.shape)
      print(y1_value==y2_value)
      print(y1_value)
      print(y2_value)
  ```

  4. run the test script

  ```
  $ python test_conv2d.py
  ```

Can anyone give some advice on how to find the problem? Thanks.

**Comment by wei-v-wang** [comment]
Any chance for you to bump your TF version? 
I tried TF v1.8.0 built with "--config=mkl" and I did not encounter this issue.

**Comment by tanhe123** [comment]
ping @wei-v-wang

**Comment by tensorflowbutler** [comment]
Nagging Assignee @poxvoculi: It has been 17 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by wei-v-wang** [comment]
Yes, thank you for trying TF v1.8.0.
